sechub-wrapper-owasp-zap/src/main/java/com/mercedesbenz/sechub/owaspzapwrapper/scan/AbstractScan.java

```diff
@@ -0,0 +1,364 @@
+// SPDX-License-Identifier: MIT
+package com.mercedesbenz.sechub.owaspzapwrapper.scan;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.zaproxy.clientapi.core.ApiResponse;
+import org.zaproxy.clientapi.core.ApiResponseElement;
+import org.zaproxy.clientapi.core.ApiResponseList;
+import org.zaproxy.clientapi.core.ClientApi;
+import org.zaproxy.clientapi.core.ClientApiException;
+
+import com.mercedesbenz.sechub.owaspzapwrapper.config.OwaspZapScanConfiguration;
+import com.mercedesbenz.sechub.owaspzapwrapper.helper.OwaspZapApiResponseHelper;
+import com.mercedesbenz.sechub.owaspzapwrapper.helper.ScanDurationHelper;
+import com.mercedesbenz.sechub.owaspzapwrapper.helper.SecHubIncludeExcludeToOwaspZapURIHelper;
+
+public abstract class AbstractScan implements OwaspZapScan {
+    private static final Logger LOG = LoggerFactory.getLogger(AbstractScan.class);
+
+    private static final int CHECK_SCAN_STATUS_TIME_IN_MILLISECONDS = 3000;
+
+    protected ClientApi clientApi;
+    protected OwaspZapScanConfiguration scanConfig;
+
+    protected String contextId;
+
+    private ScanDurationHelper scanTimingHelper;
+    private long remainingScanTime;
+
+    private SecHubIncludeExcludeToOwaspZapURIHelper includeExcludeConverter;
+
+    public AbstractScan(ClientApi clientApi, OwaspZapScanConfiguration scanConfig) {
+        this.clientApi = clientApi;
+        this.scanConfig = scanConfig;
+        this.scanTimingHelper = new ScanDurationHelper();
+        this.remainingScanTime = scanConfig.getMaxScanDurationinMillis();
+        this.includeExcludeConverter = new SecHubIncludeExcludeToOwaspZapURIHelper();
+    }
+
+    @Override
+    public void scan() {
+        try {
+            scanUnsafe();
+        } catch (ClientApiException e) {
+            LOG.error("For scan {}: An error occured while scanning! Reason: {}", scanConfig.getContextName(), e.getMessage(), e);
```

Comment:
Ahem. The scan is not working. The exception is logged. okay. but... how is the failure returned to the caller script? 

When a PDS bash script uses the owasp zap wrapper and we have here a client api exception, how shall the bash script be able to recoginze that there was a problem?